core/src/test/java/org/elasticsearch/action/admin/indices/segments/IndicesShardsStoreResponseTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.segments;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.lucene.util.CollectionUtil;
+import org.elasticsearch.Version;
+import org.elasticsearch.action.admin.indices.shards.IndicesShardsStoresResponse;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.collect.ImmutableOpenIntMap;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.transport.DummyTransportAddress;
+import org.elasticsearch.common.xcontent.*;
+import org.elasticsearch.test.ElasticsearchTestCase;
+import org.elasticsearch.transport.NodeDisconnectedException;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class IndicesShardsStoreResponseTest extends ElasticsearchTestCase {
+
+    @Test
+    public void testBasicSerialization() throws Exception {
+        ImmutableOpenMap.Builder<String, ImmutableOpenIntMap<List<IndicesShardsStoresResponse.StoreStatus>>> indexStoreStatuses = ImmutableOpenMap.builder();
+        ImmutableList.Builder<IndicesShardsStoresResponse.Failure> failures = ImmutableList.builder();
+        ImmutableOpenIntMap.Builder<List<IndicesShardsStoresResponse.StoreStatus>> storeStatuses = ImmutableOpenIntMap.builder();
+
+        DiscoveryNode node1 = new DiscoveryNode("node1", DummyTransportAddress.INSTANCE, Version.CURRENT);
+        DiscoveryNode node2 = new DiscoveryNode("node2", DummyTransportAddress.INSTANCE, Version.CURRENT);
+        List<IndicesShardsStoresResponse.StoreStatus> storeStatusList = new ArrayList<>();
+        storeStatusList.add(new IndicesShardsStoresResponse.StoreStatus(node1, 3, IndicesShardsStoresResponse.StoreStatus.Allocation.PRIMARY, null));
+        storeStatusList.add(new IndicesShardsStoresResponse.StoreStatus(node2, 2, IndicesShardsStoresResponse.StoreStatus.Allocation.REPLICA, null));
+        storeStatusList.add(new IndicesShardsStoresResponse.StoreStatus(node1, -1, IndicesShardsStoresResponse.StoreStatus.Allocation.UNUSED, new IOException("corrupted")));
```

Comment:
This `StoreStatus` represents when we fail to open the shard index due to some corruption.